full-stack-tests/presentation/src/learning-snippets/Content.test.ts

```diff
@@ -11,582 +11,1002 @@ import { getFieldByLabel, tryGetFieldByLabel } from "../Utils";
+            }],
+          }],
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure customizations have been made
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet([{ className: "BisCore:Element", id: "0x61" }]),
+          descriptor: {},
+        });
+        expect(content!.descriptor.fields.length).to.eq(20);
+        expect(tryGetFieldByLabel(content!.descriptor.fields, "User Label")).to.be.undefined;
+        expect(tryGetFieldByLabel(content!.descriptor.fields, "Parent")).to.not.be.undefined;
+        expect(tryGetFieldByLabel(content!.descriptor.fields, "Overriden Label")).to.not.be.undefined;
+      });
+
     });
 
-    it("uses `calculatedProperties` attribute", async () => {
-      // __PUBLISH_EXTRACT_START__ ContentModifier.CalculatedProperties.Ruleset
-      // The ruleset has a content rule that returns content of given input instances. There's also
-      // a content modifier that creates a calculated property for `bis.GeometricElement3d` instances.
-      const ruleset: Ruleset = {
-        id: "example",
-        rules: [{
-          ruleType: RuleTypes.Content,
-          specifications: [{
-            // load content for given input instances
-            specType: ContentSpecificationTypes.SelectedNodeInstances,
+  });
+
+  describe("Specifications", () => {
+
+    describe("ContentInstancesOfSpecificClasses", () => {
+
+      let imodel: IModelConnection;
+
+      beforeEach(async () => {
+        await initialize();
+        imodel = await SnapshotConnection.openFile("assets/datasets/Properties_60InstancesWithUrl2.ibim");
+      });
+
+      afterEach(async () => {
+        await imodel.close();
+        await terminate();
+      });
+
+      it("uses `classes` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.classes.Ruleset
+        // The specification returns content of all `bis.PhysicalModel` classes.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.Content,
+            specifications: [{
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["PhysicalModel"], arePolymorphic: false },
+            }],
+          }],
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure only the `bis.PhysicalModel` instances are selected.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+
+        expect(content!.contentSet.length).to.eq(1);
+        const calculatedField = tryGetFieldByLabel(content!.descriptor.fields, "Modeled Element");
+        expect(content!.contentSet[0].displayValues[calculatedField!.name]).to.eq("Properties_60InstancesWithUrl2");
```

Comment:
Why are we checking some random property and why is it called `calculatedField`? IMO it would be best to just check class name in instance key of the single content record.